src/expr/src/expr/expr_split_part.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use risingwave_common::array::{
+    Array, ArrayBuilder, ArrayImpl, ArrayRef, DataChunk, Utf8ArrayBuilder,
+};
+use risingwave_common::error::{Result, RwError};
+use risingwave_common::types::DataType;
+use risingwave_common::{ensure, try_match_expand};
+use risingwave_pb::expr::expr_node::{RexNode, Type};
+use risingwave_pb::expr::ExprNode;
+
+use crate::expr::{build_from_prost, BoxedExpression, Expression};
+
+#[derive(Debug)]
+pub struct SplitPartExpression {
+    return_type: DataType,
+    string_expr: BoxedExpression,
+    delimiter_expr: BoxedExpression,
+    nth_expr: BoxedExpression,
+}
+
+impl Expression for SplitPartExpression {
+    fn return_type(&self) -> DataType {
+        self.return_type.clone()
+    }
+
+    fn eval(&self, input: &DataChunk) -> Result<ArrayRef> {
+        let string_columns = self.string_expr.eval(input)?;
+        let string_columns = string_columns.as_utf8();
+
+        let delimiter_columns = self.delimiter_expr.eval(input)?;
+        let delimiter_columns = delimiter_columns.as_utf8();
+
+        let nth_columns = self.nth_expr.eval(input)?;
+        let nth_columns = nth_columns.as_int64();
```

Comment:
Here `nth` is treated as `i64`, what numerical types should we support here?

Reply:
It is `integer` (`int4` / `int32`) in PostgreSQL:
https://github.com/postgres/postgres/blob/de89d8711e2b70150a2c5b3215d99da108fbe1e8/src/include/catalog/pg_proc.dat#L3638-L3640
```
test=# select split_part(text '', text '', 3::bigint);
ERROR:  function split_part(text, text, bigint) does not exist
LINE 1: select split_part(text '', text '', 3::bigint);
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```

Reply:
thx, fixed